pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/SkylineBatchConfig.cs

```diff
@@ -0,0 +1,185 @@
+
+        private enum Attr
+        {
+            Name,
+            Created,
+            Modified
+        };
+
+
+        #region XML
+
+        public XmlSchema GetSchema()
+        {
+            return null;
+        }
+
+        public static SkylineBatchConfig ReadXml(XmlReader reader)
+        {
+            var name = reader.GetAttribute(Attr.Name);
+            DateTime created;
+            DateTime modified;
+            DateTime.TryParse(reader.GetAttribute(Attr.Created), out created);
+            DateTime.TryParse(reader.GetAttribute(Attr.Modified), out modified);
+
+            do
+            {
+                reader.Read();
+            } while (reader.NodeType != XmlNodeType.Element);
+
+            MainSettings mainSettings = null;
+            ReportSettings reportSettings = null; 
+            string exceptionMessage = null;
+            try
+            {
+                mainSettings = MainSettings.ReadXml(reader);
+                do
+                {
+                    reader.Read();
+                } while (reader.NodeType != XmlNodeType.Element);
+                reportSettings = ReportSettings.ReadXml(reader);
+            }
+            catch (ArgumentException e)
+            {
+                exceptionMessage = string.Format("\"{0}\" ({1})", name, e.Message);
+            }
+            
+            do
+            {
+                reader.Read();
+            } while (!(reader.Name == "skylinebatch_config" && reader.NodeType == XmlNodeType.EndElement));
+
+            if (exceptionMessage != null)
+                throw new ArgumentException(exceptionMessage);
+
+            return new SkylineBatchConfig(name, created, modified, mainSettings, reportSettings);
+        }
+
+        public void WriteXml(XmlWriter writer)
+        {
+            Validate();
+            writer.WriteStartElement("skylinebatch_config");
+            writer.WriteAttribute(Attr.Name, Name);
+            writer.WriteAttributeIfString(Attr.Created, Created.ToShortDateString() + " " + Created.ToShortTimeString());
+            writer.WriteAttributeIfString(Attr.Modified, Modified.ToShortDateString() + " " + Modified.ToShortTimeString());
+            MainSettings.WriteXml(writer);
+            ReportSettings.WriteXml(writer);
+            writer.WriteEndElement();
+        }
+
+
+
+        #endregion
+
+        
+
+
+        public void Validate()
+        {
+            if (MainSettings == null || ReportSettings == null)
+            {
+                throw new Exception("Configuration settings not initialized.");
+            }
+
+            if (string.IsNullOrEmpty(Name))
+            {
+                throw new ArgumentException("Please enter a name for the configuration.");
+            }
+
+            MainSettings.Validate();
+            ReportSettings.Validate();
+           
+        }
+
+        public override string ToString()
+        {
+            Validate();

```

Comment:
Validate should not be called from ToString, since ToString is not supposed to change anything.
Also, ToString() should not throw exceptions unless absolutely necessary.